src/plugins/wmr/plugin.js

```diff
@@ -90,12 +91,25 @@ export default function wmrPlugin({ hot = true } = {}) {
 			});
 
 			if (hot) {
+				const [imports] = await parse(code, id);
```

Comment:
I think we can do this in the middleware instead, since it's already parsing all of the imports?

https://github.com/preactjs/wmr/blob/3fbed71240cd1ae252c2af68192b9b0fb0cad4aa/src/wmr-middleware.js#L286

Something like:

```diff
+		const imports = new Set();

		code = await transformImports(code, id, {
			// ...
			async resolveId(spec, importer) {
				// ...
+				imports.add(spec);
				return spec;
			}
		});
+
+		code += `$IMPORT_META_HOT$.dependencies = ${JSON.stringify(Array.from(imports))}`;
```

This also is nice because it ensures they're all normalized.